lib/src/document.dart

```diff
@@ -28,13 +38,31 @@ class Document {
     this.linkResolver,
     this.imageLinkResolver,
     this.encodeHtml = true,
-  }) : extensionSet = extensionSet ?? ExtensionSet.commonMark {
-    _blockSyntaxes
-      ..addAll(blockSyntaxes ?? [])
-      ..addAll(this.extensionSet.blockSyntaxes);
-    _inlineSyntaxes
-      ..addAll(inlineSyntaxes ?? [])
-      ..addAll(this.extensionSet.inlineSyntaxes);
+    this.withDefaultBlockSyntaxes = true,
+    this.withDefaultInlineSyntaxes = true,
+  }) {
+    hasCustomInlineSyntaxes = inlineSyntaxes?.isNotEmpty == true ||
```

Comment:
Following [Effective Dart](https://dart.dev/guides/language/effective-dart/usage#prefer-using--to-convert-null-to-a-boolean-value), let's use `?? false` here, instead of `== true`.

Reply:
Fixed. This is a good one. Thanks.